### PR TITLE
small cleanups

### PR DESCRIPTION
**Pull request by tobowers**
* devLog now supports multiple non-text arguments
* CortexNext receives the step it's coming from (and memories moved to public so that they can be used)
* custom actions carried over automatically from one step to the next
* I was consistently getting some minor invalid XML, so re-inforced XML sanity in the prompt.
* test file didn't have `.test.` in it so jest wasn't finding it.

### CODE REVIEW

**Comment by kafischer** [comment]
I was wondering if next should take a action class instead - so all those built in actions aren't defined on the class and new actions aren't tied to an instance

File: core/src/cortexStep.ts [hunk 1]
```diff
@@ -195,21 +199,25 @@ export class CortexStep {
...
-    const model = `${description || `${action}`}`;
+    const model = description || action;
     const nextInstructions = [
       {
         role: "system",
         content: `
 Now, for ${this.entityName}, model ${model}.
 
-Reply in the output format: ${beginning}[[fill in]]</${action}>
+Reply in the output format: \`${beginning}[[fill in]]</${action}>\`. Double check you are returning valid XML.
```
Comment (kafischer) [inline core/src/cortexStep.ts:209]: ha, nice. funny that works well

File: core/src/cortexStep.ts [hunk 2]
```diff
@@ -195,21 +199,25 @@ export class CortexStep {
...
     const instructions = this.messages.concat(nextInstructions);
-    const nextValue = (
-      await this.processor.execute(instructions, { stop: `</${action}` })
-    ).slice(beginning.length);
+    devLog("instructions: " + instructions);
+    const resp = await this.processor.execute(instructions, {
+      stop: `</${action}`,
+    });
+    devLog("resp:", resp);
+    const nextValue = resp.slice(beginning.length);
```
Comment (kafischer) [inline core/src/cortexStep.ts:219]: if it's returning valid xml shouldn't we just parse greedily up to something like `/^[^>]*>{0,1}[^>]*>/g`

**Review by kafischer — CHANGES_REQUESTED** [review]

File: core/src/cortexStep.ts (hunk 2 above)
Comment (tobowers) [reply core/src/cortexStep.ts:219]: yeah that's good, get rid of anything in the first two > if I read that right.

**Comment by tobowers** [comment]
> I was wondering if next should take a action class instead - so all those built in actions aren't defined on the class and new actions aren't tied to an instance

Yup I definitely think it should - make using custom ones the "first class" and then include default ones in the library.

File: core/src/cortexStep.ts (hunk 2 above)
Comment (tobowers) [reply core/src/cortexStep.ts]: made that change

**Review by kafischer — APPROVED** [review]

**Pull request merged**